daemon/main.go

```diff
@@ -37,6 +39,31 @@ func createDaemon(dpuMode bool) (Daemon, error) {
 	}
 }
 
+func copyFile(src, dst string) error {
+	sourceFile, err := os.Open(src)
+	if err != nil {
+		return err
+	}
+	defer sourceFile.Close()
+
+	destinationFile, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer destinationFile.Close()
+
+	_, err = io.Copy(destinationFile, sourceFile)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func copyCNI() error {
+	return copyFile("/dpu-cni", "/opt/cni/bin/dpu-cni")
```

Comment:
this should be part of an entrypoint script tbh, this is how sriov-cni does it
I don't think daemon should be responsible for copying the binary?

Reply:
why not? What's the benefit of having an entrypoint script